dotnet/src/SemanticKernel.Core/Data/TextSearch/TextSearchServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.Data;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Extension methods to register <see cref="ITextSearch"/> for use with <see cref="IServiceCollection"/>.
+/// </summary>
+public static class TextSearchServiceCollectionExtensions
+{
+    /// <summary>
+    /// Register an <see cref="ITextSearch"/> instance with the specified service ID.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="ITextSearch"/> on.</param>
+    /// <param name="textSearch">Instance of <see cref="ITextSearch"/> to register.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    public static IServiceCollection AddTextSearch(
```

Comment:
I'm not super convinced about the value of the first two overloads here.  I think what's more useful would be something like AddBingSearch that just takes the options for that.  After all someone can just call AddKeyedSingleton themselves here.  We are not really doing anything extra.

Reply:
Yeah, I agree. I'll move these to the Web plugin